experiments/AtmosLES/squall_line.jl

```diff
@@ -0,0 +1,270 @@
+    IceParameterSet(),
+    RainParameterSet(),
+    SnowParameterSet(),
+)
+const param_set = EarthParameterSet(microphys)
+
+using Dierckx
+ClimateMachine.init(parse_clargs = true)
+
+function init_squall_line!(problem, bl, state, aux, localgeo, t, args...)
+    FT = eltype(state)
+    spl_tinit, spl_qinit, spl_uinit, spl_vinit, spl_pinit = args[1]
+    # interpolate data
+    (x, y, z) = localgeo.coord
+    data_t = FT(spl_tinit(z))
+    data_q = FT(spl_qinit(z))
+    data_u = FT(spl_uinit(z))
+    data_v = FT(spl_vinit(z))
+    data_p = FT(spl_pinit(z))
+    u = data_u
+    v = data_v
+    w = FT(0)
+    if z >= 14000
+        data_q = FT(0)
+    end
+    θ_c = 3.0
+    rx = 10000.0
+    ry = 1500.0
+    rz = 1500.0
+    xc = 0.5 * (FT(0) + FT(0))
+    yc = 0.5 * (FT(0) + FT(5000))
+    zc = 2000.0
+    cylinder_flg = 0.0
+    r = sqrt(
+        (x - xc)^2 / rx^2 +
+        cylinder_flg * (y - yc)^2 / ry^2 +
+        (z - zc)^2 / rz^2,
+    )
+    Δθ = 0.0
+    if r <= 1.0
+        Δθ = θ_c * (cospi(0.5 * r))^2
+    end
+    θ_liq = data_t + Δθ
+    ts = PhaseNonEquil_pθq(param_set, data_p, θ_liq, PhasePartition(FT(data_q)))
+    T = air_temperature(ts)
+    ρ = air_density(ts)
+    e_kin = FT(1 / 2) * FT((u^2 + v^2 + w^2))
+    e_pot = gravitational_potential(bl.orientation, aux)
+    E = ρ * total_energy(e_kin, e_pot, ts)
+    state.ρ = ρ
+    state.ρu = SVector(ρ * u, ρ * v, FT(0))
+    state.ρe = E
+    state.moisture.ρq_tot = ρ * data_q
+    state.moisture.ρq_liq = FT(0)
+    state.moisture.ρq_ice = FT(0)
+    return nothing
+end
+
+function read_sounding()
+    #read in the original squal sounding
+    fsounding = open(joinpath(@__DIR__, "../sounding_gabersek.dat"))
+    #fsounding  = open(joinpath(@__DIR__, "../sounding_gabersek_3deg_warmer.dat"))
+    sounding = readdlm(fsounding)
+    close(fsounding)
+    (nzmax, ncols) = size(sounding)
+    if nzmax == 0
+        error("SOUNDING ERROR: The Sounding file is empty!")
+    end
+    return (sounding, nzmax, ncols)
+end
+
+function spline_int()
+
+    # ----------------------------------------------------
+    # GET DATA FROM INTERPOLATED ARRAY ONTO VECTORS
+    # This driver accepts data in 6 column format
+    # ----------------------------------------------------
+    (sounding, _, ncols) = read_sounding()
+
+    # WARNING: Not all sounding data is formatted/scaled
+    # the same. Care required in assigning array values
+    # height theta qv    u     v     pressure
+    zinit, tinit, qinit, u_init, v_init, pinit = sounding[:, 1],
+    sounding[:, 2],
+    0.001 .* sounding[:, 3],
+    sounding[:, 4],
+    sounding[:, 5],
+    sounding[:, 6]
+    spl_tinit = Spline1D(zinit, tinit; k = 1)
+    spl_qinit = Spline1D(zinit, qinit; k = 1)
+    spl_uinit = Spline1D(zinit, u_init; k = 1)
+    spl_vinit = Spline1D(zinit, v_init; k = 1)
+    spl_pinit = Spline1D(zinit, pinit; k = 1)
+    return spl_tinit, spl_qinit, spl_uinit, spl_vinit, spl_pinit#, spl_rhoinit, spl_ppiinit, spl_thetainit
```

Comment:
```suggestion
    return (t=spl_tinit, q=spl_qinit, u=spl_uinit, v=spl_vinit, p=spl_pinit)
```
Let's use a `NamedTuple` here to improve readability and reduce the risk of mixing up the order.